File: src/libs/ThirdParty.VBSuppliers/Properties/ExpediaRapid/SerializableClasses/Book/BookRequest.vb
Imports Newtonsoft.Json

Namespace ExpediaRapid.SerializableClasses.Book
    Public Class BookRequest

        <JsonProperty("affiliate_reference_id")>
        Public Property AffiliateReferenceId As String

        <JsonProperty("hold")>
        Public Property Hold As Boolean

        <JsonProperty("rooms")>
        Public Property Rooms As New List(Of BookRequestRoom)

        <JsonProperty("payments")>
        Public Property Payments As New List(Of Payment)

        <JsonProperty("affiliate_metadata")>
        Public Property AffiliateMetadata As String

    End Class

End Namespace

File: src/libs/ThirdParty.CSSuppliers/ExpediaRapid/SerializableClasses/Book/BookRequest.cs
namespace ThirdParty.CSSuppliers.ExpediaRapid.SerializableClasses.Book
{
    using System.Collections.Generic;
    using Newtonsoft.Json;

    public class BookRequest
    {

        [JsonProperty("affiliate_reference_id")]
        public string AffiliateReferenceId { get; set; }

        [JsonProperty("hold")]
        public bool Hold { get; set; }

        [JsonProperty("rooms")]
        public List<BookRequestRoom> Rooms { get; set; } = new List<BookRequestRoom>();

        [JsonProperty("payments")]
        public List<Payment> Payments { get; set; } = new List<Payment>();

        [JsonProperty("affiliate_metadata")]
        public string AffiliateMetadata { get; set; }

    }

}